pkg/controllermanager/deployer/deployer.go

```diff
@@ -570,7 +570,7 @@ func (deployer *Deployer) populateLocalizations(sub *appsapi.Subscription, base
 				continue
 			}
 
-			if len(replicas) < clusterIndex {
+			if len(replicas) <= clusterIndex {
```

Comment:
This index starts from 0. Thus it should be `<` here.

Reply:
supposing bindingClusters are cluster0, cluster1, cluster2, cluster3, and replicas are [1, 1, 1], when clusterIndex is 3, the following code will panic in "replicas[clusterIndex]"

```golang
loc.Spec.Overrides = []appsapi.OverrideConfig{
				{
					Name:  "dividing scheduling replicas",
					Value: fmt.Sprintf(`[{"path":%q,"value":%d,"op":"replace"}]`, feedOrder.ReplicaJsonPath, replicas[clusterIndex]),
					Type:  appsapi.JSONPatchType,
				},
			}
```

Reply:
> supposing `bindingClusters` are cluster0, cluster1, cluster2, cluster3, and replicas are [1, 1, 1]

Here the length of `bindingClusters` and replicas should be equal. And they are supposed to be equal. Would you share some clues on this? Thanks.

The current fix will bring panic as well. If `len(bindingClusters)=4`  and `len(replicas)=3`, then `len(replicas) <= 3 (the last cluster index)`, this leads to panic for `replicas[3]`.

Reply:
If len(bindingClusters)=4 and len(replicas)=3, then len(replicas) <= 3 (the last cluster index), it will not go to the code "replicas[3]" and continue to process the next feed

Reply:
Normally, the two are not equal, but when the scheduler is abnormal, it will cause the entire system to be unavailable, so protection should be done here.

Reply:
I got it. Nice catch.

This is a warning logic. So it is fine to have `<=` here.